packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmUtils.kt

```diff
@@ -203,3 +235,46 @@ private fun <T : RealmObject> processListMember(
     }
     return list
 }
+
+fun genericRealmCoreExceptionHandler(message: String, cause: RealmCoreException): Throwable {
```

Comment:
Are we sure that this mapping always apply?

Reply:
Anyway, we should probably not map `RealmCoreOtherException` to `IllegalArgumentException` as this is actually an internal core runtime error. Confusion maybe stems from that error being raised when thawing a deleted object, but that is exactly an faulty implementation in core.

Reply:
This function only applies to immutable schema Realms, another kind of Realms would need another one.